File: CS/XpoXtraReports/Orderline.cs
using System;
using DevExpress.Xpo;

namespace XpoXtraReports
{

    public class OrderLine : XPObject
    {
        public OrderLine(Session session)
            : base(session)
        {
            // This constructor is used when an object is loaded from a persistent storage.
            // Do not place any code here.
        }

        public override void AfterConstruction()
        {
            base.AfterConstruction();
            // Place here your initialization code.
        }


        #region Property Order
        private Order _Order;
        [Association("Order-Lines", typeof(OrderLine))]
        public Order Order
        {
            get { return this._Order; }
            set { SetPropertyValue<Order>("Order", ref this._Order, value); }
        }
        #endregion 
        

        #region Properties
        private string _Item;
        private decimal _Cost;
        public string Item{
            get { return this._Item; }
            set { SetPropertyValue<string>("Item", ref this._Item, value); }
        }
        public decimal Cost {
            get { return this._Cost; }
            set { SetPropertyValue<decimal>("Cost", ref this._Cost, value); }
        }
        #endregion         
    }
}

File: VB/XpoXtraReports/Orderline.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace XpoXtraReports

	Public Class OrderLine
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
			' This constructor is used when an object is loaded from a persistent storage.
			' Do not place any code here.
		End Sub

		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
			' Place here your initialization code.
		End Sub


		#Region "Property Order"
		Private _Order As Order
		<Association("Order-Lines", GetType(OrderLine))> _
		Public Property Order() As Order
			Get
				Return Me._Order
			End Get
			Set(ByVal value As Order)
				SetPropertyValue(Of Order)("Order", Me._Order, value)
			End Set
		End Property
		#End Region 


		#Region "Properties"
		Private _Item As String
		Private _Cost As Decimal
		Public Property Item() As String
			Get
				Return Me._Item
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Item", Me._Item, value)
			End Set
		End Property
		Public Property Cost() As Decimal
			Get
				Return Me._Cost
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue(Of Decimal)("Cost", Me._Cost, value)
			End Set
		End Property
		#End Region         
	End Class
End Namespace